lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if !array1 && !array2
+    return true
+  elsif !array1 || !array2
+    return false
```

Comment:
This is a handy thing to learn when working with logic: https://thepugautomatic.com/2012/09/de-morgans-laws-in-programming/

`!(array1 && array2) == (!array1 || !array2)` being equivalent means they can be written and convey the same outcomes.